### PR TITLE
Add a CMakeLists, tests, and pipeline to automatically test with Github actions

### PR DESCRIPTION
**Pull request by skyedeving**
Adds CMakeLists, tests, and a pipeline to test with Github actions. I've used these mostly to validate my PR's.

# Writing tests
See Catch2 documentation.
Use [noci] tag to have it not run during CI.

# Github Actions
Features two different workflows
* Test workflow runs tests on any push: https://github.com/skyedeving/AO2-Client/actions?query=workflow%3Atest
* Build workflow builds an artifact on push to master that can be downloaded for testing: https://github.com/skyedeving/AO2-Client/actions?query=workflow%3Abuild

# Test
```sh
mkdir cbuild &&  cd cbuild
cmake ..

# Compile everything
make

# Compile individual
make Attorney_Online
make test # Requires catch2

# Run test
./test/test

# Run tests with tag and excluding tag
./test/test [apng] ~[aopacket] 
```

### CODE REVIEW

**Comment by oldmud0** [comment]
This looks good, but I can only merge this pull request once.

**Comment by skyedeving** [comment]
Sounds good, the branch has pretty much stabilized. I should've split this up into two separate PR's back when it would've been easier to do so.

**Comment by Crystalwarrior** [comment]
No clue how this works so I'll leave it up to @oldmud0 to decide when to merge

**Review by stonedDiscord — APPROVED** [review]

File: test/CMakeLists.txt
Comment (oldmud0) [inline test/CMakeLists.txt:4]: I know you want to make the test build go super fast, but is manually adding the dependent source files really best practice? I would think that adding the rest of the AO sources with a CMake include would only add fifteen seconds of build time.

Comment (skyedeving) [reply test/CMakeLists.txt:4]: I actually have no idea regarding what's best practice. There is a second Github workflow which creates a test build although that should take care of that: https://github.com/AttorneyOnline/AO2-Client/blob/6f5e933c18e0dcd9d5ee7f1ebb51adb70e868a09/.github/workflows/build.yml

**Review by oldmud0 — APPROVED** [review]
Looks great, I'll give the Windows artifact a go once I get back to my computer again, and if it is then we can merge!

**Review by oldmud0 — APPROVED** [review]
Okay. Let's do it.

**Pull request merged**
